h2/src/main/org/h2/value/Transfer.java

```diff
@@ -594,6 +608,14 @@ public Value readValue() throws IOException {
             return ValueDouble.get(readDouble());
         case Value.FLOAT:
             return ValueFloat.get(readFloat());
+        case Value.ENUM: {
+            int len = readInt();
+            String[] enumerators = new String[len];
+            for (int i = 0; i < len; i++) {
+                enumerators[i] = readString();
+            }
+            return ValueEnum.get(enumerators, readInt());
+        }
```

Comment:
which will require some ickiness here, in that you'll need to construct a ValueEnum that does not know what it's complete set of enumerators are.
We face a similar problem with ValueLob and we get around it by passing in a DataHandler interface (really bad name).
We could do something similar with ValueEnum, passing in an interface which allows it request the full set of enumerators for itself on-demand.